---
float32 last_beta